Modified request_ip function to actually return the ip instead of "expression"

aoby opened the pull request:
Setup

``` ruby
irb(main):002:0> request = OpenStruct.new
=> #<OpenStruct>
irb(main):003:0> request.ip = '1.1.1.1'
=> "1.1.1.1"
```

Old method:

``` ruby
irb(main):004:0> defined? request ? request.ip : nil
=> "expression"
```

New method:

``` ruby
irb(main):005:0> defined?(request) ? request.ip : nil
=> "1.1.1.1"
```

aoby commented:
Could also do it this way:

``` ruby
irb(main):030:0> defined?(request) ? request.ip : nil
=> "1.1.1.1"
```

laurelfan commented on pegasus/helpers/prize_helpers.rb:
```diff
@@ -1,7 +1,8 @@
+  request.ip
 end
 
 def claim_prize_code(type, email, purpose, params={})
```
do we expect the ip address to be included in params most of the time?

aoby replied:
It's not currently included, and since this is called from process_forms the request.ip is always 127.0.0.1. Perhaps we should pass in the ip address from the form.

aoby commented:
Updated to set the claimed_ip to the ip in the prize selection form.
Added a script to update all existing prize ips from 'expression' to the ip from the associated form.

Tested locally

laurelfan commented on pegasus/forms/hoc_survey_prize_2015.rb:
```diff
@@ -9,9 +9,10 @@ def self.normalize(data)
     result
   end
 
-  def self.process(data)
```
what's the difference between `process` and `process_`?

aoby replied:
See https://github.com/code-dot-org/code-dot-org/blob/865ca031618ff16b02b8291bde2dd5776b7e48bc/bin/process_forms#L53-L57

process_ receives the entire form, while process only receives the parsed json data field. 

As to why they are named that way, I can't tell you. I'd prefer process_form & process_data or something similar but I don't think it's worth updating all the existing forms.

aoby replied:
Also I called the parameter row because I was copying another [form](https://github.com/code-dot-org/code-dot-org/blob/1b19ed45e17897881d00fcdd931c489aea9d05e4/pegasus/forms/professional_development_workshop.rb#L59-L59), but I think 'form' is more clear so I'll update that.

laurelfan commented:
LGTM
